### PR TITLE
MAINT: Remove code using vulture confidence 60

### PR DESCRIPTION
**Pull request by larsoner**
Turns out we had a bunch of dead code. We also had a bunch of unused global vars in our test files which this caught!

This also makes me think that we should maybe add a test that checks the inventory of the testing dataset to make sure all files are used. Given the removals here it's possible that some no longer are, and it's either a bug and we should add them back somehow or we should remove them from the testing dataset...

### CODE REVIEW

**Comment by hoechenberger** [comment]
Wow this is **massive**

File: doc/sphinxext/gen_commands.py [hunk 1]
```diff
@@ -106,6 +108,17 @@ def generate_commands_rst(app=None):
     _replace_md5(str(out_fname))
 
 
+def _replace_md5(fname):
+    """Replace a file based on MD5sum."""
+    # adapted from sphinx-gallery
+    assert fname.endswith(".new")
```
Comment (drammock) [inline doc/sphinxext/gen_commands.py:114]: this won't work for Path objects, is that OK?

Comment (larsoner) [reply doc/sphinxext/gen_commands.py:114]: Must be, this is just copy-pasted :slightly_smiling_face:

**Comment by larsoner** [comment]
@drammock this one should be good to go now!

File: mne/time_frequency/csd.py [hunk 2]
```diff
@@ -1007,7 +1007,7 @@ def csd_array_multitaper(
     n_times = len(times)
     n_fft = n_times if n_fft is None else n_fft
 
-    window_fun, eigvals, mt_adaptive = _compute_mt_params(
+    window_fun, eigvals, adaptive = _compute_mt_params(
```
Comment (drammock) [inline mne/time_frequency/csd.py:1010]: yikes, this was probably a bug

**Review by drammock — APPROVED** [review]
startling how much unused code in there. Some of it seems useful enough that I'm wondering if it should either get promoted to public API, or spun off into a utils-like section of `mne-incubator`. I'm thinking of things like

- `_is_scaled_mri_subject`
- `_copy_channel`
- `_ensure_image_in_surface_RAS`
- `_pair_grad_sensors_*` (for when info absent, e.g., projectors)

Feel free to merge if you disagree; I don't have time to do it myself ATM so I don't want to be a blocker.

**Comment by larsoner** [comment]
>  Some of it seems useful enough that I'm wondering if it should either get promoted to public API, or spun off into a utils-like section of mne-incubator

Perhaps. I'm happy to leave it out for now and see if people need it and assume YAGNI until then. Plus we don't know the extent to which those snippets of code still work with our codebase at this point since they haven't been tested. It's easy enough to find in the history especially knowing the lines were removed in this `vulture` PR, though, if we do want them.

**Pull request merged**